doc/rest-authentication.md

```diff
@@ -0,0 +1,106 @@
+# Glusterd2 REST API authentication
+
+Glusterd2 REST API authentication is based on [JWT](https://jwt.io). If REST
+authentication is enabled then each client request should include
+`Authorization` header as example below.
+
+    Authorization: bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnbHVzdGVyY2xpIiwiaWF0IjoxNTMxNzI5NzQyLCJleHAiOjE1MzE3Mjk3NTJ9._WIwO7PrHUHIT62SdzfkyNjqD1GEgX2cYqN8ACZCtaw
+
+**Note**: REST Authentication can be disabled by adding `restauth=false`
+in Glusterd2 config file(Default path is
+`/var/lib/glusterd2/glusterd2.toml` in case of rpm installation)
+
+## Default user
+
+When `glusterd2` starts for the first time, it creates the
+`$GLUSTERD2_STATEDIR/auth` file which will contain the secret. If
+"gluster" user group is available in the system then this `auth` file
```

Comment:
Do we mention how to create "gluster" user group?

Reply:
I don't think we need to do this. This is simple unix knowledge. In any case, if installed from packages, the package should take care of creating the users/groups. I don't think any gluster packages do this yet.

Reply:
It may be worthwhile to point out this is a unix group interacting with file permissions and not some special gluster thing in that case. The word group and phrase "in that machine" alone may not hint at that strongly enough.